python/fusion_engine_client/messages/configuration.py

```diff
@@ -399,3 +414,164 @@ def __str__(self):
 
     def calcsize(self) -> int:
         return len(self.pack())
+
+
+class InterfaceID(NamedTuple):
+    type: TransportType = TransportType.INVALID
+    index: int = 0
+
+
+class OutputInterfaceConfig(NamedTuple):
+    output_interface: InterfaceID = InterfaceID()
+    stream_indices: List[int] = []
+
+
+_InterfaceIDConstructRaw = Struct(
+    "type" / AutoEnum(Int8ul, TransportType),
+    "index" / Int8ul,
+    Padding(2)
+)
+_InterfaceIDConstruct = NamedTupleAdapter(InterfaceID, _InterfaceIDConstructRaw)
+
+
+class _OutputInterfaceConfigConstruct(Adapter):
+    """!
+    Adapter to handle setting `num_streams` implicitly.
+    """
+    def __init__(self):
+        super().__init__(Struct(
+            "output_interface" / _InterfaceIDConstruct,
+            "num_streams" / Int8ul,
+            Padding(3),
+            "stream_indices" / Array(this.num_streams, Int8ul),
+        ))
+
+    def _decode(self, obj, context, path):
+        return OutputInterfaceConfig(obj.output_interface, obj.stream_indices)
+
+    def _encode(self, obj, context, path):
+        return {
+            "output_interface": obj.output_interface,
+            "num_streams": len(obj.stream_indices),
+            "stream_indices": obj.stream_indices
+        }
+
+
+class SetOutputInterfaceStreamsMessage(MessagePayload):
+    """!
+    @brief Configure the set of output streams enabled for a given output interface.
```

Comment:
It might help to have usage examples for these more complicated set streams and set config classes.